src/components/button-toggle-group/button-toggle-group.js

```diff
@@ -0,0 +1,161 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import css from './../../utils/css';
+
+import InputLabel from './../../utils/decorators/input-label';
+import InputValidation from './../../utils/decorators/input-validation';
+
+import tagComponent from '../../utils/helpers/tags';
+
+/**
+ * An inoput group widget.
+ *
+ * == How to use an ButtonToggleGroup in a component:
+ *
+ * In your file
+ *
+ *   import ButtonToggleGroup from 'carbon/lib/components/button-toggle-group';
+ *
+ * To render an ButtonToggleGroup:
+ *
+ *   <ButtonToggleGroup validations=[]>
+ *     <ButtonToggle />
+ *   <ButtonToggleGroup />
+ *
+ * @class ButtonToggleGroup
+ * @constructor
+ * @decorators {InputLabel,InputValidation}
+ */
+const ButtonToggleGroup = InputLabel(InputValidation(
+class ButtonToggleGroup extends React.Component {
+  static propTypes = {
+    /**
+     * Children elements
+     *
+     * @property children
+     * @type {Node}
+     */
+    children: PropTypes.node.isRequired,
+
+    /**
+     * Custom className
+     *
+     * @property className
+     * @type {String}
+     */
+    className: PropTypes.string,
+
+    /**
+     * Selected value from children components
+     *
+     * @property value
+     * @type {String}
+     */
+    value: PropTypes.string
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (this.messageExists() && (nextProps.value !== this.props.value)) {
+      let contentChanged = false;
+
+      if (!this.state.valid && this.validate(nextProps.value)) {
+        this.setState({ valid: true });
+        contentChanged = true;
+      }
+
+      if (this.state.warning && !this.warning(nextProps.value)) {
+        this.setState({ warning: false });
+        contentChanged = true;
+      }
+
+      if (this.state.info && !this.info(nextProps.value)) {
+        this.setState({ info: false });
+        contentChanged = true;
+      }
+
+      if (contentChanged) {
+        this._handleContentChange();
+      }
+    }
+  }
+
+  /**
+   * Main Class getter
+   *
+   * @method mainClasses
+   * @return {void}
+   */
+  get mainClasses() {
+    return classNames('carbon-button-toggle-group', this.props.className, css.input);
+  }
+
+  /**
+   * Provides an extension point to add additional properties for the containing field.
+   *
+   * @method fieldProps
+   * @return {Object} Field props
+   */
+  get fieldProps() {
+    const fieldProps = {
+      className: 'common-input__field',
+      onFocus: this._handleFocus,
+      onBlur: this._handleBlur
```

Comment:
does this support focus and blur?

Reply:
Yes, for keyboard navigation. If we don't handle focus and blur, a user tabbing through controls doesn't trigger or clear the message.